docs/usage/configuration/config_documentation.md

```diff
@@ -3170,6 +3170,10 @@ Options for each entry include:
 * `client_secret`: oauth2 client secret to use. May be omitted if
   `client_secret_jwt_key` is given, or if `client_auth_method` is 'none'.
 
+* `client_secret_file`: path to the oauth2 client secret to use. With that
+   it's not necessary to leak secrets into the config file itself.
+   Mutually exclusive with `client_secret`.
```

Comment:
For the `[signing_key` we seem to use `signing_key_path`](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#signing-keys) (as opposed to `signing_key_file`). I'd recommend being consistent with that.